reactor-core-micrometer/src/test/java/reactor/core/observability/micrometer/TimedSchedulerTest.java

```diff
@@ -324,4 +329,65 @@ void workerSchedulePeriodicallyIsCorrectlyMetered() throws InterruptedException
 				+ testScheduler.submittedPeriodicInitial.count()
 				+ testScheduler.submittedPeriodicIteration.count(), "completed tasks == sum of all timer counts");
 	}
+
+	@Test
+	void pendingScheduleRemovedOnScheduleRejection() {
+		CountDownLatch cdl = new CountDownLatch(1);
```

Comment:
Instead of using a latch, consider creating a disposed `Scheduler` (e.g. `Schedulers.newSingle("testScheduler"`).

Reply:
As noticed in the comment below, feel free to ignore this suggestion once a check is added for the metric being non-zero.